### PR TITLE
Movie api

### PR DESCRIPTION
**Pull request by egutten**
User stories solved for in this branch (in the context of creating a new movie entry):

1. As a user, I want to see a dropdown list of movies based on what I type in the title box, so that I do not have to remember the whole title myself.

2. As a user, I want to select the title from the dropdown and have my selection populate the title input, so that I do not have the type the title myself.

3. As an engineer, I do not want the user to be able to make duplicate movies with the same title, so that my website has variability.

4. As a user, if the movie title I selected is already taken I want to be notified of the issue, so I can pick another title.

### CODE REVIEW

File: routes/movies.js [hunk 1]
```diff
@@ -45,10 +47,13 @@ router.post("/", middleware.isLoggedIn, function(req, res){
 	req.body.movie.image = req.sanitize(req.body.movie.image);
 	Movie.create(req.body.movie, function(err, newMovie){
 		if(err){
-			console.log(err);
-			req.flash("error", "Something went wrong...");
-			res.render("movies/new");
-		} else {
+			if(err.code == 11000){
```
Comment (mgutten) [inline routes/movies.js:48]: Did you intend to leave the node api request in the code base?

File: public/javascript/avengers.js [hunk 2]
```diff
@@ -17,5 +17,39 @@ $(function() {
...
+					 $("#results").append(option);
+					 $("#results").trigger("chosen:updated");
+				});
+			 }
+		  },
+        });
+	});
+});
 
+$("#results").on("click", "li.select", function(){
```
Comment (mgutten) [inline public/javascript/avengers.js:43]: Nice!

File: public/javascript/avengers.js [hunk 3]
```diff
@@ -17,5 +17,39 @@ $(function() {
...
+          url: 'https://movie-database-imdb-alternative.p.rapidapi.com/?page=1&r=json&s=' + value,
+          type: 'GET',
+          dataType: 'json',
+          beforeSend: setHeader,
+		  error: function() { alert('boo!'); },
+          success: function(msg) { 
+			 $("#results").show();
+			 if(value==$(that).val() && msg["Error"] == undefined) {
+				 msg["Search"].forEach (function(e){
+					 var option = document.createElement("li");
```
Comment (mgutten) [inline public/javascript/avengers.js]: Not critical - should we move the HTML creation to its own function?

File: public/javascript/avengers.js [hunk 4]
```diff
@@ -17,5 +17,39 @@ $(function() {
...
+					 option.innerHTML = e["Title"];
+					 option.classList.add("select");
+					 $("#results").append(option);
+					 $("#results").trigger("chosen:updated");
```
Comment (mgutten) [inline public/javascript/avengers.js]: What’s the purpose of the trigger function? 🤷🏼‍♂️

**Review by mgutten — COMMENTED** [review]
Great job!! I’m on the edge of my seat to see how it works!!

File: routes/movies.js (hunk 1 above)
Comment (egutten) [reply routes/movies.js:48]: I did not intend to leave the backend api packages in, and I deleted them this morning in the latest commit.

File: public/javascript/avengers.js (hunk 3 above)
Comment (egutten) [reply public/javascript/avengers.js]: Done last night, with the help of a genius meepies.

File: public/javascript/avengers.js (hunk 4 above)
Comment (egutten) [reply public/javascript/avengers.js]: The trigger function allows the api query to refresh every time the user types different content into the title input.

**Comment by egutten** [comment]
I just updated the pull request. Should be good to go, but let me know!

**Comment by mgutten** [comment]
Very clean, LGTM (looks good to me)!

**Pull request merged**